pkg/apis/operator/v1alpha1/tektonconfig_validation.go

```diff
@@ -33,6 +33,14 @@ func (tc *TektonConfig) Validate(ctx context.Context) (errs *apis.FieldError) {
 		return nil
 	}
 
+	// disallow to update the targetNamespace
+	if apis.IsInUpdate(ctx) {
```

Comment:
do we need this if condition ?

Reply:
working fine without if condition, should we remove it?

Reply:
I had removed if condition and few tests started failing so we should keep the if condition

Reply:
> I had removed if condition and few tests started failing so we should keep the if condition

If tests are failing, but usecase is working, then I would say we need to fix the tests

Reply:
validation test for normal context is failing as `existingTC := apis.GetBaseline(ctx).(*TektonConfig)` is getting `nil` and for inupdate context it gets tektonconfig so we should keep if condition

Reply:
I think, we need this if condition, to know is it a update request or new request

Reply:
This condition make sure the "request" was a UPDATE and not at creation, … I wonder if it also takes into account PATCH (if not, we need to add patch there as well).

So yes, we should keep that condition *and* maybe check if PATCH is required as well.

Reply:
aah ohk, make sense. Thanks for the explanation

Reply:
I think these are the only option

```
func IsInCreate(ctx context.Context) bool
func IsInDelete(ctx context.Context) bool
func IsInSpec(ctx context.Context) bool
func IsInStatus(ctx context.Context) bool
func IsInStatusUpdate(ctx context.Context) bool
func IsInUpdate(ctx context.Context) bool
```

Reply:
yes, there is no option for patch

Reply:
Perfect, we are good to go then it seems